backend/src/main/java/com/htecgroup/skynest/filter/CustomAuthenticationFilter.java

```diff
@@ -0,0 +1,73 @@
+package com.htecgroup.skynest.filter;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.htecgroup.skynest.exception.UserException;
+import com.htecgroup.skynest.exception.UserExceptionType;
+import com.htecgroup.skynest.model.request.UserLoginRequest;
+import com.htecgroup.skynest.security.SecurityConstants;
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+
+@AllArgsConstructor
+@Log4j2
+public class CustomAuthenticationFilter extends UsernamePasswordAuthenticationFilter {
+
+    private final AuthenticationManager authenticationManager;
+    private final UserDetailsService userDetailsService;
+    private final PasswordEncoder passwordEncoder;
+    @Override
+    public Authentication attemptAuthentication(HttpServletRequest request, HttpServletResponse response) throws AuthenticationException {
+        UserLoginRequest credentials = new UserLoginRequest();
+        try {
+            credentials = new ObjectMapper().readValue(request.getInputStream(), UserLoginRequest.class);
+            log.info("Here, with help of object mapper we map the value of the request to UserLoginRequest class");
```

Comment:
You don't need to explain what is done here. The purpose of log statement is to tell us what has been done in the app in case debugging is needed. 
Better log statement will be to say which user is trying to authenticate

Reply:
Got it, I changed all the logs as well.